Documentation/building/build-guidelines.md

```diff
@@ -0,0 +1,47 @@
+Build guidelines
+================
+###Goal
+
+It is difficult for devs, who are new to the corefx repo, to reason about the build and understand how to implement new features, or otherwise update the repo's build.  The primary goal is to provide a repo which is easier to reason about.  A secondary goal, would be to help the dev implement their changes to the build (without getting in their way) such that they are confident they are following our guidelines.
+
+- Keep it simple
+- Keep it clean
+- Make it maintainable (secondary)
+
+### Keep it simple
+
+We don't want multiple ways to do the same thing.  There is a support cost and mind-share cost associated with having multiple ways to achieve something.  It's ok to have insider tricks, things for power-users who understand the default experience but want to optimize it for their development patterns.
```

Comment:
Yes, I'm not proposing we remove msbuild, build.cmd, run.cmd; just that we minimize it so that the root of the repo only has run.cmd.  msbuild is still available from there, but the hope is that this people look at the repo, see run.cmd, and wonder, what does this thing do...?  Then we use run to teach them how to interact with the repo.  build.cmd is still an option, but less obvious as it will be moved out of the root along with all of the other cmd files (except run.cmd).